### ISSUE TITLE
Modernization (Structure, Coding Style etc...)

### ISSUE THREAD

**Issue by a4lg** [ssdeep-project/ssdeep#2]
ssdeep needs modernization. Because this program has long history, some part of the code is getting old. As a part of major program restructuring, I'm doing some modernization.

* Coding style changes  
  To make program consistent, easy to read and easy to contribute, coding style of ssdeep will be updated to relatively major one.
* Deprecating some language features  
  ssdeep uses some features on C/C++ language that are inconsistent and make the program confusing. Uses of those features will be removed.
* Portability improvements  
  Except some library part, I'm going to make the program portable as possible.
* Readability improvements

# Coding Style Changes

* 1 indent == 4 spaces
* no tabs
* similar to K&R
* basically limited to 78 columns (but with exceptions)

# Deprecateing Language Features

* Alternative operators (`and`, `or` and `not`)  
  Use `&&`, `||` and `!`.

# Portability

* Portable integer types  
  Some uses of non-portable types (`uint8_t` and `uint64_t`) will be changed to make the program portable. Note that `uint32_t` argument of `fuzzy_hash_buf` will be preserved for now.
